src/uwtools/tests/fixtures/testdriver.yaml

```
@@ -0,0 +1,4 @@
+testdriver:
+  execution:
+    executable: "{{ testdriver.rundir }}/testdriver"
+  rundir: /path/to/rundir/
```

```suggestion
testdriver: null
```
Given that `TestDriver` doesn't actually use anything from its config, and that its schema supports any JSON object, maybe this can be simplified to avoid giving the impression that anything more complex is happening. I tried this suggestion on your branch and it seemed to be fine (all tests passed).